pkg/knowledge_base2/properties/property.go

```diff
@@ -0,0 +1,74 @@
+package properties
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+)
+
+type (
+	SharedPropertyFields struct {
+		DefaultValueTemplate *template.Template `json:"default_value" yaml:"default_value"`
+		ValidityChecks       []PropertyValidityCheck
+	}
+
+	PropertyValidityCheck struct {
+		template *template.Template
+	}
+	ValidityCheckData struct {
+		Properties construct.Properties
+		Value      any
+	}
+)
+
+func ParsePropertyRef(value any, ctx knowledgebase.DynamicContext, data knowledgebase.DynamicValueData) (any, error) {
```

Comment:
Shouldn't this always return a `PropertyRef`, not `any`?

Reply:
yeah changing this